src/toxcore/dht/packet/dht_request.rs

```diff
@@ -264,9 +272,70 @@ impl ToBytes for DhtPkAnnounce {
     }
 }
 
+/** Hardening nodes request of DHT Request packet.
+
+Length    | Content
+--------- | -------------------------
+`1`       | `0x30`
+`1`       | `0x02`
+`rest`    | ignored
+
+*/
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct HardeningRequest;
+
+impl FromBytes for HardeningRequest {
+    named!(from_bytes<HardeningRequest>, do_parse!(
+        tag!("\x30") >>
+        tag!("\x02") >>
+        rest >>
+        (HardeningRequest)
+    ));
+}
+
+impl ToBytes for HardeningRequest {
+    fn to_bytes<'a>(&self, buf: (&'a mut [u8], usize)) -> Result<(&'a mut [u8], usize), GenError> {
+        do_gen!(buf,
+            gen_be_u8!(0x30) >>
+            gen_be_u8!(0x02)
+        )
+    }
+}
+
+/** Hardening nodes response of DHT Request packet.
+
+Length    | Content
+--------- | -------------------------
+`1`       | `0x30`
+`1`       | `0x03`
+`rest`    | ignored
+
+*/
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct HardeningResponse;
+
+impl FromBytes for HardeningResponse {
+    named!(from_bytes<HardeningResponse>, do_parse!(
+        tag!("\x30") >>
+        tag!("\x03") >>
+        rest >>
```

Comment:
It's not necessary to parse `rest` since we don't use it and don't require `eof`.

Reply:
That's a big question what shall we do here

Reply:
I suggest to leave `rest` but we have to mark with comments what is going on. Also structs like HardeningResponse and HardeningRequest should be commented, that they are not used and they are depricated.

Reply:
done